lib/cloudant.js

```diff
@@ -493,14 +500,14 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   };
 
   if (id) {
-    self.updateAttributes(model, id, data, {}, function(err, docs) {
+    self.update(model, {id: id}, data, {}, function(err, docs) {
       if (err && err.statusCode !== 404) return cb(err);
       else if (err && err.statusCode === 404) {
         // strip out the `_rev` property before creation
         delete data._rev;
         self.create(model, data, {}, createHandler);
       } else {
-        return cb(err, self.fromDB(model, mo, docs), {isNewInstance: false});
+        return cb(err, docs[0], {isNewInstance: false});
```

Comment:
Ok, I think I understand now. On the previous PR #123, we just stripped of `_rev` instead of throwing an error. Talked with @jannyHou and we came to a conclusion of throwing an error is the better way to go if the user specifies `_rev` on a create operation.

We shall create another story/PR for it and keep it separate from this one.